CAIPs/caip-74.md

```diff
@@ -1,144 +1,271 @@
+CACAO proposes a way to leverage [varsig](https://github.com/ChainAgnostic/varsig) and [multidid](https://github.com/ChainAgnostic/multidid/) as well as IPLD to create a common representation for various different object-capability formats, such as SIWE and UCAN. The IPLD representation contains common fields shared between these format. In addition this CAIP also registers varsig codes for both SIWE + ReCap and UCAN. 
 
 ## Motivation
 
-"Sign-in with X" is a way for a user to authenticate into a service, and provide authorization. In essense, it is a signature of a well-formed payload.
+There has been a proliferation of ways to create object-capabilities in the web3 space. Most notably [Sign-in with Ethereum](https://eips.ethereum.org/EIPS/eip-4361), [UCAN](https://github.com/ucan-wg/spec), [Sign-in with X](https://chainagnostic.org/CAIPs/caip-122), and [ReCap](https://eips.ethereum.org/EIPS/eip-5573). All of these approaches share similar characteristics such as an issuer, audience, signature, etc. However, they are structured quite differently and have different constraints when it comes to how they are serialized and signed. Having a common representation strategy in IPLD can make it easier to interface with these various formats.
 
-We could see this as a stepping point for a _rich_ capability-based authorization system.
+## Specification
 
-In order to do this, we would like to have a standardized IPLD-based representation of the payload and the signature, that together comprise a capability.
+The specification consists of two main things, an IPLD schema that describes the data structure of a CACAO, and algorithms to convert SIWE(x), ReCap, and UCAN messages into and out of this data structure.
 
-## Specification
+### Container IPLD schema
 
-### Container format
+The container schema described SHOULD be encoded using the `dag-cbor` IPLD codec.
 
-We start construction with declaring a container format, that represents a signed payload.
-It should contain meta-information, payload and signatures. For reference let's call such container _CACAO_ (for Chain Agnostic CApability Object).
-We use [IPLD schema language](https://ipld.io/docs/schemas/) to describe the format.
-Reminder, unless a field is marked `optional`, it is mandatory.
+```verilog
+
+type Prinicpal Bytes // a multidid
+type Varsig Bytes
+type Resource String // URL
+type Ability String // e.g. crud/create
+
+type NB { String : Any }
+type Abilities { Ability : [NB] }
```

Comment:
I think we need to specify the semantics of this array: it can be interpreted as a logical `OR` or `AND`. Here's a contrived example:

```json
{
  "https://example.com/blog/posts": {
    "crud/create": [
      {"day": "friday"},
      {"status": "draft"}
    ]
  }
}
```

One can interpret this as either
* "You can post drafts to the blog on Fridays"
* "You can post drafts to the blog, or post any status on Fridays"

In UCAN <=0.9, the second looks something like...

```json
[
  {
    "with": "https://example.com/blog/posts",
    "can": "crud/create",
    "nb": {"day": "friday"}
  },
  {
    "with": "https://example.com/blog/posts",
    "can": "crud/create",
    "nb": {"status": "draft"}
  }
]
```

...which has a very clear intention that this is an `OR`. Using this logic, to make it an AND, you need to put all keys in a single entry:

```json
{
  "https://example.com/blog/posts": {
    "crud/create": [
      {
        "day": "friday",
        "status": "draft"
      }   
    ]
  }
}
```

I'm fairly certain that without specifying this, we open ourselves to confused deputies based on misinterpreting the semantics from the source capability format.

Reply:
Good point. Will address this!

Reply:
Added an example. Let me know if this fully addresses your concern!